forager_frontend/src/node/src/DatasetList.js

```diff
@@ -16,56 +20,134 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import fromPairs from "lodash/fromPairs";
 import toPairs from "lodash/toPairs";
 
-const endpoints = fromPairs(toPairs({
-  getDatasets: "get_datasets",
-  createDataset: "create_dataset",
-}).map(([name, endpoint]) => [name, `${process.env.REACT_APP_SERVER_URL}/api/${endpoint}`]));
+const JOB_STATUS_INTERVAL = 5000;
+
+const endpoints = fromPairs(
+  toPairs({
+    getDatasets: "get_datasets",
+    createDataset: "create_dataset",
+    startInference: "start_model_inference",
+    inferenceStatus: "model_inference_status",
+  }).map(([name, endpoint]) => [
+    name,
+    `${process.env.REACT_APP_SERVER_URL}/api/${endpoint}`,
+  ])
+);
 
 const DatasetList = () => {
   const [datasets, setDatasets] = useState([]);
+  const [unfinishedJobs, setUnfinishedJobs] = useState({});
```

Comment:
Seems like there's a race condition in both async functions where `setUnfinishedJobs` is called: `getJobStatus` could wipe out a new job that is added while it is running and `createDataset` could wipe out job status updates that are fetched while it is waiting on the network request. This is not a huge problem (it's just the dataset list) but I think an easy fix is turning `unfinishedJobs` into a `useReducer` - see what we do for `categoryCounts` and `knnImages` in App.js.